src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Workspace.cs

```diff
@@ -529,19 +529,19 @@ public async Task<T> WriteAsync<T>(Func<IWorkspace, Task<T>> action, Cancellatio
 
         cancellationToken = CancellationTokenSource.CreateLinkedTokenSource(_unloadCancellationToken, cancellationToken).Token;
 
-        await WhenInitialized(cancellationToken);
+        await WhenContextCreated(cancellationToken);
 
         return await ExecuteUnderLockAsync(_ => action(this), cancellationToken);
     }
 
-    private async Task WhenInitialized(CancellationToken cancellationToken)
+    private async Task WhenContextCreated(CancellationToken cancellationToken)
     {
         Verify.NotDisposed(this);
 
         await _joinableTaskFactory.RunAsync(async () =>
         {
-            // We only have build data if we also have evaluation data, so this implies both have been integrated.
-            await _hasBuildData.Task.WithCancellation(cancellationToken);
+            // Ensure we have received enough data to create the context.
```

Comment:
This will hang if the data that will eventually cause this TaskCompletionSource to be set needs the UI thread.

Reply:
That was fixed in https://github.com/dotnet/project-system/pull/8492

Reply:
Another similar fix https://github.com/dotnet/project-system/pull/8501